File: 1Urte desberdinetako materiala/Ethazi 25-26/multi_intefaz/src/main/java/model/Writer.java

package model;


public class Writer {
     private int id;
     private String izena;

    public Writer(int id, String izena) {
        this.id = id;
        this.izena = izena;
    }

    public Writer() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getIzena() {
        return izena;
    }

    public void setIzena(String izena) {
        this.izena = izena;
    }

    @Override
    public String toString() {
        return "Writer{" + "id=" + id + ", izena=" + izena + '}';
    }
    

}

File: PMD 25-26/Erronka/app/src/main/java/com/example/erronka/model/Writer.kt
package com.example.erronka.model

/**class Writer {
    var id: Int = 0
    var izena: String? = null

    constructor(id: Int, izena: String?) {
        this.id = id
        this.izena = izena
    }

    constructor()

    override fun toString(): String {
        return "Writer{" + "id=" + id + ", izena=" + izena + '}'
    }
}**/


class Writer {

    private var id: Int = 0
    private var izena: String = ""

    // Constructor con parámetros
    constructor(id: Int, izena: String) {
        this.id = id
        this.izena = izena
    }

    // Constructor vacío
    constructor()

    // Getters y Setters
    fun getId(): Int = id
    fun setId(id: Int) { this.id = id }

    fun getIzena(): String = izena
    fun setIzena(izena: String) { this.izena = izena }

    // toString
    override fun toString(): String {
        return "Writer(id=$id, izena=$izena)"
    }
}